auth/detect/internal/externalaccount/info.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package externalaccount
+
+import (
+	"runtime"
+	"strings"
+	"unicode"
+)
+
+var (
+	// version is a package internal global variable for testing purposes.
+	version = runtime.Version
+)
+
+// versionUnknown is only used when the runtime version cannot be determined.
+const versionUnknown = "UNKNOWN"
+
+// goVersion returns a Go runtime version derived from the runtime environment
+// that is modified to be suitable for reporting in a header, meaning it has no
+// whitespace. If it is unable to determine the Go runtime version, it returns
+// versionUnknown.
+func goVersion() string {
```

Comment:
Any interest in using the `gax.GoVersion`? https://pkg.go.dev/github.com/googleapis/gax-go/v2#pkg-variables

Not sure if it would be good to take the dep though...

Reply:
Nope for that exact reason haha. We may eventually want/need to do it but it would be unfortunate. Right now this module overall does depend on the gRPC stack, but the detect tree does not. And Go is smart enough to only realize used package level deps. Only the grpctransport package currently brings in that dep. If we ever made a gax v3 it would be nice to separate large external deps to their own package.

Reply:
For sure, makes sense! thanks.